MaterialXLoader: added invert implementation

beersandrew opened the pull request:
- allow invert to be a number, vec 2|3|4, or color, subtracting the value from an optionally inputted amount field (which is defaulted to all parts 1 of the input type)

Related issue: #29432

**Description**

Implemented the invert node for the MaterialXLoader

*This contribution is funded by [Needle](https://needle.tools)*

beersandrew committed:
feat: added invert implementation

- allow invert to be a number, vec 2|3|4, or color, subtracting the value from an optionally inputted amount field (which is defaulted to all parts 1 of the input type)

sunag commented on examples/jsm/loaders/MaterialXLoader.js:
```diff
@@ -44,6 +44,37 @@ const mx_power = ( in1, in2 = float( 1 ) ) => pow( in1, in2 );
 const mx_atan2 = ( in1 = float( 0 ), in2 = float( 1 ) ) => atan2( in1, in2 );
 const mx_timer = () => timerLocal();
 const mx_frame = () => frameId;
+const mx_invert = ( in1, amount = null ) => {
```
Could you use the approach below instead of checking node values? Type conversions are done automatically prioritizing large vectors, if the input of `in1` is `vec3` it will return a `vec3`, and so we'll assume that `amount` can be classified as a node abstraction not just for constants.

```js
const mx_invert = ( in1, amount = float( 1 ) ) => sub( amount, in1 );
```

beersandrew replied:
Thanks, that's much better! Updated.

sunag commented:
The implementation of https://github.com/mrdoob/three.js/issues/29440 should be simple, in case you want to implement this PR? 

```js
const extract_x = ( in1 ) => in1.x;
```

The index parameter seems redundant here based on the function name...

beersandrew commented:
> The implementation of #29440 should be simple, in case you want to implement this PR?
> 
> ```js
> const extract_x = ( in1 ) => in1.x;
> ```
> 
> The index parameter seems redundant here based on the function name...

I think it's needed, here the `name` is only an identifier with `extract` being the node name(https://github.com/beersandrew/MaterialX/blob/separateNodeExample/resources/Materials/Examples/StandardSurface/extract.mtlx#L8)

not sure if this works? but this is the route i'll try (should work for colorN & vectorN)
```js
const extract = ( in1, index ) => in1[index];
```

beersandrew committed:
feat: update invert

- simplified invert function
- removed unneeded imports
- moved invert to math nodes

sunag commented:
I think it's better to use element in this case, with `[ index ]` the input values ​​should be just integer js numbers not nodes.
```js
const extract = ( in1, index ) => in1.element( index );
```
